test/jdk/com/sun/jdi/PopFramesTest.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Call popFrames() on threads in various states not covered
+ *          well by other tests. Most notably, this test includes
+ *          test cases for a suspended but unmounted virtual thread.
+ *          It is mostly for testing for OpaqueFrameException and
+ *          NativeMethodException.
+ *
+ * @run build TestScaffold VMConnection TargetListener TargetAdapter
+ * @run compile -g PopFramesTest.java
+ * @run driver PopFramesTest SLEEP_NATIVE
+ * @run driver PopFramesTest LOOP_NATIVE
+ * @run driver PopFramesTest SLEEP_PRENATIVE
+ * @run driver PopFramesTest LOOP_PRENATIVE
+ * @run driver PopFramesTest SLEEP_NONATIVE
+ * @run driver PopFramesTest LOOP_NONATIVE
+ */
+import com.sun.jdi.*;
+import com.sun.jdi.event.*;
+import java.util.*;
+
+/*
+ * There are six test modes covered by this test:
+ *   SLEEP_NATIVE
+ *   LOOP_NATIVE
+ *   SLEEP_PRENATIVE
+ *   LOOP_PRENATIVE
+ *   SLEEP_NONATIVE
+ *   LOOP_NONATIVE
+ *
+ * SLEEP:     the debuggee blocks in Thread.sleep()
+ * LOOP:      the debuggee sits in a tight loop
+ * NATIVE:    there is a native frame within the set of frames to pop.
+ * PRENATIVE: there is a native frame before the set of frames to pop.
+ * NONATIVE:  there is no native frame (purposefully) present in the stack.
+ *
+ * In all cases the thread is suspended and errors such as IllegalArgumentException
+ * and InvalidStackFrameException should not happen. The popFrames() calls  should
+ * either pass, or produce OpaqueFrameException or NativeMethodException.
+ *
+ * Call stacks for each test mode (and expected result):
+ *  - Note in all cases the popMethod() frame is the frame passed to popFrames()..
```

Comment:
Nit: Is there an extra dot at the end it was intentional?
       The lines 54-58 are inconsistent with dots at the end.